pkg/webhook/default_server/uniteddeployment/validating/uniteddeployment_validation.go

```diff
@@ -0,0 +1,166 @@
+		case "StatefulSet", "ReplicasSet":
+		default:
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("template", "subsetType"), spec.Template.SubsetType.Kind, fmt.Sprintf("must be 'StatefulSet' or 'ReplicasSet'")))
+		}
+	}
+
+	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*spec.Replicas), fldPath.Child("replicas"))...)
+	if spec.Selector == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("selector"), ""))
+	} else {
+		allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(spec.Selector, fldPath.Child("selector"))...)
+		if len(spec.Selector.MatchLabels)+len(spec.Selector.MatchExpressions) == 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("selector"), spec.Selector, "empty selector is invalid for statefulset"))
+		}
+	}
+
+	if spec.Template.StatefulSet == nil && spec.Template.ReplicaSet == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("template", "statefulSet"), ""))
+	}
+	if spec.Template.StatefulSet != nil && spec.Template.ReplicaSet != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("template", "statefulSet"), spec.Template.StatefulSet, "only support one kind of subset"))
+	}
+
+	selector, err := metav1.LabelSelectorAsSelector(spec.Selector)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("selector"), spec.Selector, ""))
+	} else {
+		if spec.Template.StatefulSet != nil {
+			template := spec.Template.StatefulSet.Spec.Template
+			coreTemplate, err := convertPodTemplateSpec(&template)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Root(), spec.Template, fmt.Sprintf("Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec failed: %v", err)))
+				return allErrs
+			}
+			allErrs = append(allErrs, appsvalidation.ValidatePodTemplateSpecForStatefulSet(coreTemplate, selector, fldPath.Child("template", "statefulSet"))...)
+		}
+		if spec.Template.ReplicaSet != nil {
+			template := spec.Template.ReplicaSet.Spec.Template
+			coreTemplate, err := convertPodTemplateSpec(&template)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Root(), spec.Template, fmt.Sprintf("Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec failed: %v", err)))
+				return allErrs
+			}
+			allErrs = append(allErrs, appsvalidation.ValidatePodTemplateSpecForReplicaSet(coreTemplate, selector, *spec.Replicas, fldPath.Child("template", "replicasSet"))...)
+		}
+	}
+
+	var sumReplicas int32
+	count := 0
+	for _, subset := range spec.Topology.Subsets {
+		if len(subset.Name) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("topology", "subset", "name"), ""))
+		}
+
+		if errs := apimachineryvalidation.NameIsDNSLabel(subset.Name, false); len(errs) > 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("topology", "subset", "name"), subset.Name, fmt.Sprintf("invalid subset name %s", strings.Join(errs, ", "))))
+		}
+
+		if spec.Replicas == nil || subset.Replicas == nil {
```

Comment:
This logic looks weird.  If spec.Replicas is unset, we should assume total replica is the default (0 or 1 as I suggest in the comment) and do the sum check.

Reply:
done